meta-ros2/recipes-devtools/ros-sdk-env/ros-sdk-env_1.0.bb

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) Stephen Street stephen@redrocketcomputing.com
+
+SUMMARY = "Add ROS2 SDK enviroment variable"
+HOMEPAGE = "https://github.com/meta-ros"
+SECTION = "devel"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/ros-sdk-env:"
+
+SRC_URI += "file://generate-colcon-default"
+
+inherit python3native ros_opt_prefix
+
+FILES:${PN} = "${base_prefix}/environment-setup.d/ros-sdk-env.sh"
+
+S = "${WORKDIR}"
+
+# This was pulled from meta-ros/meta-ros2/classes/ros_ament_cmake.bbclass
+PYTHON_SOABI_ARCH = "${TUNE_ARCH}-${TARGET_OS}"
+PYTHON_SOABI_ARCH_SUFFIX = "-gnu"
+
+# The suffix is already included in TARGET_OS
+PYTHON_SOABI_ARCH_SUFFIX:arm = ""
+
+# Another exception is i686 TUNE_ARCH in dunfell and newer with this change:
+# https://git.openembedded.org/openembedded-core/commit/?h=dunfell&id=6beab388e73b3ac6157650855a6c1fb1d71e8015
+PYTHON_SOABI_ARCH:i686 = "i386-${TARGET_OS}"
+
+PYTHON_SOABI = "cpython-${@d.getVar('PYTHON_BASEVERSION').replace('.', '')}${PYTHON_ABI}-${PYTHON_SOABI_ARCH}${PYTHON_SOABI_ARCH_SUFFIX}"
+
+do_install () {
+    echo "export ROS_VERSION=${ROS_VERSION}" > ${S}/ros-sdk-env.sh
+    echo "export ROS_PYTHON_VERSION=${ROS_PYTHON_VERSION}" >> ${S}/ros-sdk-env.sh
+    echo "export ROS_DISTRO=${ROS_DISTRO}" >> ${S}/ros-sdk-env.sh
+    echo "export PYTHONPATH="'$OECORE_NATIVE_SYSROOT'"/usr/lib/python${PYTHON_BASEVERSION}/site-packages:"'$OECORE_TARGET_SYSROOT'"${ros_base_prefix}/lib/python${PYTHON_BASEVERSION}/site-packages"  >> ${S}/ros-sdk-env.sh
+    echo "export AMENT_PREFIX_PATH="'$OECORE_TARGET_SYSROOT'"${ros_base_prefix}" >> ${S}/ros-sdk-env.sh
+    echo "export PYTHON_SOABI=${PYTHON_SOABI}.so" >> ${S}/ros-sdk-env.sh
```

Comment:
`PYTHON_SOABI` should not have a .so prefix:

```
builder@7f008748f080:/build$ bitbake-getvar PYTHON_SOABI -r rmw-implementation
WARNING: You have included the meta-virtualization layer, but 'virtualization' has not been enabled in your DISTRO_FEATURES. Some bbappend files may not take effect. See the meta-virtualization README for details on enabling virtualization support.
#
# $PYTHON_SOABI
#   set /build/../work/src/ros/meta-ros2/classes/ros_ament_cmake.bbclass:15
#     "cpython-${@d.getVar('PYTHON_BASEVERSION').replace('.', '')}${PYTHON_ABI}-${PYTHON_SOABI_ARCH}${PYTHON_SOABI_ARCH_SUFFIX}"
PYTHON_SOABI="cpython-312-aarch64-linux-gnu"
```

Reply:
See CMAKE_PYTHON_SOABI at line 40.  This value should be used in the --cmake-args.

Reply:
I understand, but you're setting it to `${PYTHON_SOABI}`**.so**. The `.so` needs to go.

Reply:
I picked this you from net somewhere as required. I do not understand of this environment variable effects python interpreter execution. @robwoolley has a difference approach which I also do not understand.  I think we need clarity, idea?